### PR TITLE
setup azure pipeline build specs for intern branches

### PR DESCRIPTION
**Pull request by acchiang**
## Overview

Update the existing azure-pipeline.yml file to use a general Azure Pipelines agent pool. 
Builds are now also triggered by `interns/*` branches.

### CODE REVIEW

**Comment by acchiang** [comment]
image

**Comment by acchiang** [comment]
requesting review from interns and Mark first before sending to coaches (not sure what the process is yet)!

File: azure-pipelines.yml [hunk 1]
```diff
@@ -8,7 +8,10 @@
 
 name: $(date:yyyyMMdd)$(rev:.r)
 
-pool: MsGraphBuildAgentsWindows
+pool:
+  vmImage: 'windows-latest'
+  steps:
+  - script: echo hello from Windows
```
Comment (acchiang) [inline azure-pipelines.yml:14]: we don't have access to this [agent pool](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=azure-devops&tabs=yaml%2Cbrowser) that is used for the original repo, so i used the default Azure Pipelines pool here with a Windows vmImage.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -18,6 +21,7 @@ pr:
       - feature/*
       - task/*
       - fix/*
+      - interns/*
```
Comment (acchiang) [inline azure-pipelines.yml:24]: @AyuAi let me know if there are any other branch-naming conventions we will be using for interns? I think it's just `interns/` prefix on all our branches right? 

FYI this means the build will be triggered for each PR and push to a branch with the `interns/` prefix

**Review by gracet9182 — APPROVED** [review]

**Review by omersayshi — APPROVED** [review]

**Review by lhao03 — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (ayubun) [reply azure-pipelines.yml:24]: @acchiang Everything should start with interns/* except for our deploys to “production,” which we will use “dev” for ^^

Basically, the pipeline is our branches -> interns/dev -> dev. Though, we wont merge into interns/dev unless it passes the checks, so interns/dev -> dev merges should theoretically have no more checks, right?

I think it wouldn’t hurt to add dev to this list just incase 👀

**Review by ayubun — CHANGES_REQUESTED** [review]

**Review by jerryxihe — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (acchiang) [reply azure-pipelines.yml:24]: Don't worry, dev is on line 20 (it was already in the original azure_pipeline.yml file)!

**Review by ayubun — APPROVED** [review]

**Pull request merged**